File: backend/gms/src/main/java/com/g4/gms/dto/ReminderRequest.java
package com.g4.gms.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ReminderRequest {
    @NotEmpty(message = "Title cannot be empty")
    private String title;
    
    @NotEmpty(message = "Message cannot be empty")
    private String reminderMessage;
    
    @NotEmpty(message = "Reminder date cannot be empty")
    private String reminderDate;
    
    private String scheduleId;
} 

File: frontend-mobile/app/src/main/java/com/example/GarbageMS/models/ReminderRequest.kt
package com.example.GarbageMS.models

import java.util.Date

data class ReminderRequest(
    val title: String,
    val reminderMessage: String,
    val reminderDate: Date,
    val scheduleId: String,
    val fcmToken: String? = null  // Optional FCM token field
) 